### PR TITLE
Add missing CellML 1.0 supported subset MathML functions

### LINKED ISSUE #139: Add missing trig functions
- [ ] add to language in `maths.py`
- [x] add parsing rules in CompactSyntaxParser
- [ ] add to function list `MATHML` in `actions.py`
- [ ] add to sympy function list in `actions.py` 
- [ ] add floor/ceiling tests to `test_mathml_evaluation.py`
- [ ] add trig tests to `test_mathml_evaluation.py`

### PR DESCRIPTION
**Pull request by MichaelClerx**
Fixes #139

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -9,6 +9,7 @@ addons:
 install:
   # Needed to run setup.py
   - pip install cython numpy
+  - pip install git+https://github.com/ModellingWebLab/cellmlmanip.git@more-trig-functions
```
Comment (jonc125) [inline .travis.yml:12]: This will need to change back before merging :)

File: fc/language/expressions/trig.py [hunk 2]
```diff
@@ -0,0 +1,201 @@
...
+    """Cotangent."""
+    _op = 'cot'
+
+    def _ex(self, a):
+        return 1 / math.tan(a)
+
+
+class ArcSec(TrigExpression):
+    """Inverse secant."""
+    _op = 'sec'
```
Comment (jonc125) [inline fc/language/expressions/trig.py:133]: I think this is wrong?!

File: fc/language/expressions/trig.py [hunk 3]
```diff
@@ -0,0 +1,201 @@
...
+
+    def _ex(self, a):
+        return math.acos(1 / a)
+
+
+class ArcCsc(TrigExpression):
+    """Inverse cosecant."""
+    _op = 'csc'
```
Comment (jonc125) [inline fc/language/expressions/trig.py:141]: Same here

File: fc/language/expressions/trig.py [hunk 4]
```diff
@@ -0,0 +1,201 @@
...
+
+    def _ex(self, a):
+        return math.asin(1 / a)
+
+
+class ArcCot(TrigExpression):
+    """Inverse cotangent."""
+    _op = 'cot'
```
Comment (jonc125) [inline fc/language/expressions/trig.py:149]: And here, and all the ones below!

File: test/output/mathml_evaluation/generate.py [hunk 5]
```diff
@@ -0,0 +1,136 @@
...
+# Stop after abs
+#ops = ops[:16]
+
+np.seterr(all='ignore')
+
+eol = '\n'
+with open('outputs_combined_pos.csv', 'w') as f:
+    f.write('# All outputs for positive inputs' + eol)
+    for i in range(-300, 325, 25):
+        # The code below has some funny bits, to more closely emulate whatever
```
Comment (jonc125) [inline test/output/mathml_evaluation/generate.py:112]: It would have been the C++ implementation I think!

**Review by jonc125 — APPROVED** [review]
Just some minor copy & paste errors spotted. I don't think they affect any results!

File: fc/language/expressions/trig.py (hunk 2 above)
Comment (MichaelClerx) [reply fc/language/expressions/trig.py:133]: Woops. How did that pass tests...

Comment (MichaelClerx) [reply fc/language/expressions/trig.py]: Thanks! Have added tests and fixes

File: .travis.yml (hunk 1 above)
Comment (MichaelClerx) [reply .travis.yml:12]: Does that mean I need to ask maurice to do a cellmlmanip release before we can merge this?

Comment (jonc125) [reply .travis.yml:12]: Yes please @MauriceHendrix  :)

**Pull request merged**